codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -368,20 +400,12 @@ class KotlinJmespathExpressionVisitor(
     private val Shape.isEnumMap: Boolean
         get() = this is MapShape && ctx.model.expectShape(value.target).isEnum
 
-    private fun ensureNullGuard(shape: Shape?, expr: String, elvisExpr: String? = null): String =
-        if (shape?.isNullable == true) {
-            buildString {
-                append("?.$expr")
-                elvisExpr?.let { append(" ?: $it") }
-            }
-        } else {
-            ".$expr"
+    private fun addNullGuard(shape: Shape?, expr: String, elvisExpr: String? = null): String =
+        buildString {
+            append("?.$expr")
+            elvisExpr?.let { append(" ?: $it") }
         }
```

Comment:
**Question**: Why are we adding null guards to potentially-non-null expressions? We [made the codegen visitor null-aware](https://github.com/awslabs/smithy-kotlin/pull/842) in response to a waiters bug and I worry that regressing that behavior could introduce new bugs.